sequencer/node/node.go

```
@@ -184,11 +196,23 @@ func NewNode( /*mode Mode, */ cfg *config.Node, version string) (*Node, error) {
 		scryptN = keystore.LightScryptN
 		scryptP = keystore.LightScryptP
 	}
-	keyStore = keystore.NewKeyStore(
-		cfg.Coordinator.EthClient.Keystore.Path,
-		scryptN,
-		scryptP,
-	)
+	keyStore = keystore.NewKeyStore(cfg.Coordinator.EthClient.Keystore.Path, scryptN, scryptP)
+
+	isEmpty, err := isDirectoryEmpty(cfg.Coordinator.EthClient.Keystore.Path)
+	if err != nil {
+		log.Fatalf("Error checking keystore directory: %v", err)
+	}
+
+	if isEmpty {
+		// Create a new account if keystore is empty
+		account, err := keyStore.NewAccount(cfg.Coordinator.EthClient.Keystore.Password)
+		if err != nil {
+			log.Fatalf("Error creating new account: %v", err)
+		}
+		log.Infof("New account created: %s", account.Address.Hex())
```

we should either kill the app here or update the forger address in cfg.Coordinator.ForgerAddress:
```go
cfg.Coordinator.ForgerAddress = account.Address
```
otherwise the app crashes anyway, I recommend the latter case

---

It will still only update forger address in node file but not in cfg.toml file. So I suggest that we update it manually in the cfg file itself.

---

yes we should update in cfg manually as well but I think we shouldn't let the app crash. It should either exit gracefully or internally update the forger address and continue